src/components/ModalWithForm/ModalWithForm.js

```diff
@@ -13,6 +14,31 @@ export default function ModalWithForm({
   signinError,
   signupError,
 }) {
+  // when press 'esc' close popup
+  useEffect(() => {
+    const exitEsc = (e) => {
+      if (e.key === "Escape") {
+        onClose();
+      }
+    };
+    document.addEventListener("keydown", exitEsc);
+
+    return () => document.removeEventListener("keydown", exitEsc);
+  }, [isOpen, onClose]);
+
+  // when click on modal then close popup
+  useEffect(() => {
+    if (isOpen) {
+      document
+        .querySelector(".modal_type_opened")
```

Comment:
* please, delete the overlay listener. See here https://disk.yandex.ru/i/C8yfzQvcyACfXQ . It's not the way it should be. You add listeners again and again to the same elements. This can cause a **`memory leak`**

You need to change it somehow in a more React way


please, add ` onClick={handleOverlay} ` to the `div` in line 43


```jsx


  // here is the overlay handler
  const handleOverlay = (e) => {
    if (e.target === e.currentTarget) {
        onClose();
    }
  }

  return (
    <div className={`modal ${name} ${isOpen ? "modal_type_opened" : ""}`} onClick={handleOverlay} >
```

Reply:
Thank you. I fixed it.